akka-actor/src/main/scala/akka/pattern/CircuitBreakersRegistry.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2022 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.pattern
+
+import java.util.concurrent.ConcurrentHashMap
+
+import scala.concurrent.duration.{ DurationLong, MILLISECONDS }
+
+import akka.actor.{ ExtendedActorSystem, Extension, ExtensionId, ExtensionIdProvider }
+import akka.pattern.internal.CircuitBreakerTelemetryProvider
+import akka.util.ccompat.JavaConverters._
+
+/**
+ * Companion object providing factory methods for Circuit Breaker which runs callbacks in caller's thread
+ */
+object CircuitBreakersRegistry extends ExtensionId[CircuitBreakersRegistry] with ExtensionIdProvider {
+
+  /**
+   * Is used by Akka to instantiate the Extension identified by this ExtensionId,
+   * internal use only.
+   */
+  override def createExtension(system: ExtendedActorSystem): CircuitBreakersRegistry =
+    new CircuitBreakersRegistry(system)
+
+  /**
+   * Returns the canonical ExtensionId for this Extension
+   */
+  override def lookup: ExtensionId[_ <: Extension] = CircuitBreakersRegistry
```

Comment:
two more needed for it to work well from Java:
```
  override def get(system: ActorSystem): CircuitBreakersRegistry = super.get(system)

  override def get(system: ClassicActorSystemProvider): CircuitBreakersRegistry = super.get(system)
```

Reply:
Done ✅